transform/coroutines.go

```diff
@@ -600,11 +600,11 @@ func (c *coroutineLoweringPass) lowerFuncsPass() {
 			continue
 		}
 
-		if len(fn.normalCalls) == 0 {
-			// No suspend points. Lower without turning it into a coroutine.
+		if len(fn.normalCalls) == 0 && fn.fn.FirstBasicBlock().FirstInstruction().IsAAllocaInst().IsNil() {
```

Comment:
I don't think alloca instructions are always at the start of the entry block. I think it would be safer to check the entire entry block for alloca instructions, just in case some are not the first instruction.

(Technically they can be anywhere in the function but we check in other places that this isn't possible).

Reply:
If there are any alloca instructions that are not at the start of a block, the coroutine lowering pass crashes iirc.

Reply:
IIRC it crashes if they are not in the entry block. but I think they are allowed if they are not directly at the start of the entry block.